clang/test/CodeGenHipStdPar/rdc-does-not-enable-hipstdpar.cpp

```diff
@@ -0,0 +1,11 @@
+// Check that if we are compiling with fgpu-rdc amdgpu-enable-hipstdpar is not
+// passed to CC1, to avoid eager, per TU, removal of potentially accessible
+// functions.
+
+// RUN: %clang -### --hipstdpar --offload-arch=gfx906 %s -nogpulib -nogpuinc \
+// RUN:   2>&1 | FileCheck -check-prefix=NORDC %s
+// NORDC: {{.*}}"-mllvm" "-amdgpu-enable-hipstdpar"
+
+// RUN: %clang -### --hipstdpar --offload-arch=gfx906 %s -nogpulib -nogpuinc -fgpu-rdc \
+// RUN:   2>&1 | FileCheck -check-prefix=RDC %s
+// RDC-NOT: {{.*}}"-mllvm" "-amdgpu-enable-hipstdpar"
```

Comment:
```suggestion
// RDC-NOT: -amdgpu-enable-hipstdpar
```

-NOT checks are hazardous and should be as permissive as possible 
```suggestion
// RDC-NOT: {{.*}}"-mllvm" "-amdgpu-enable-hipstdpar"
```

Reply:
In this case it actually has to be "-mllvm" because we only care about it not being passed to the initial from source, per TU compilation; forming the check as you suggest would (erroneously) match the (intentional) passing of the argument via `-plugin-opt`, when we do the final lowering from bitcode. This merely tests/validates the change we did in `HIPAMDToolChain::addClangTargetOptions`.